src/Microsoft.NetCore.Analyzers/CSharp/Performance/CSharpMutableStructsShouldNotBeUsedForReadonlyFields.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.Analyzers.Performance;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Performance
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpMutableStructsShouldNotBeUsedForReadonlyFieldsAnalyzer : MutableStructsShouldNotBeUsedForReadonlyFieldsAnalyzer
+    {
+        protected override void RegisterDiagnosticAction(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(NodeAction, SyntaxKind.FieldDeclaration);
+        }
+
+        private static void NodeAction(SyntaxNodeAnalysisContext context)
```

Comment:
In general, we strongly recommend not using SyntaxNodeAction in this repo. Majority of analyzers should fall under the below 2 buckets:

1. Analyze the attributes/properties of a symbols: Prefer RegisterSymbolAction
2. Analyze executable code: Prefer RegisterOperationAction

SyntaxNodeActions are only required for analyses related to syntax nodes that are not part of any symbol's declaration OR executable code, which is extremely rare for an analyzers in this repo.

Reply:
📝 Note that syntax node actions are not inherently bad. However, the `GetSymbolInfo` method is very expensive, and the cost can be avoided by using one of the other approaches. If you need to operate on symbols (as opposed to just syntax), symbol and operation actions are more efficient. In some cases, operation actions also simplify the work required to support both C# and VB.

Reply:
@mareklinka regarding the "rewrite" of this analyzer, you can have a look at this PR I started: https://github.com/dotnet/roslyn-analyzers/pull/3405/files